test/e2e/setup/setup.go

```diff
@@ -112,7 +114,18 @@ func GeneratePasswordForUser(mdbu mdbv1.MongoDBUser, ctx *e2eutil.Context, names
 		SetField(passwordKey, password).
 		Build()
 
-	return password, e2eutil.TestClient.Create(context.TODO(), &passwordSecret, &e2eutil.CleanupOptions{TestContext: ctx})
+	err = e2eutil.TestClient.Create(context.TODO(), &passwordSecret, &e2eutil.CleanupOptions{TestContext: ctx})
```

Comment:
I'm not 100% sure about this. I think we should decide if this test wants a new password or not, or we'll depend on some kind of previous state.

Each test's executing environment should be explicit, and not have any hidden state. In the case of this snippet, we will be adding a conditioning state to the test run, possible making it not deterministic.

Reply:
very good point about consistency, I think there's no harm in adding some unique string to the name of each secret instead of re-using existing ones. What do you think?

Reply:
I think that's a much better approach.